src/core/cgroup-bpf-ctx.c

```diff
@@ -1,22 +1,83 @@
 /* SPDX-License-Identifier: LGPL-2.1+ */
 
+#include "bpf-program.h"
 #include "cgroup-bpf-ctx.h"
 #include "cgroup.h"
 #include "memory-util.h"
 #include "strv.h"
 #include "string-util.h"
 #include "path-util.h"
+#include "strv.h"
+
+int bpffs_program_from_string(const char *str, enum bpf_attach_type *ret_attach_type, char **ret_bpffs_path) {
+        _cleanup_strv_free_ char **parts = NULL;
+        _cleanup_free_ char *dup = NULL;
+        int attach_type;
+
+        assert(str);
+        assert(ret_bpffs_path);
+        assert(ret_attach_type);
+
+        parts = strv_split(str, ":");
```

Comment:
if there are more than two components you ignore every single one, but the rest, that doesn't look right.

strv_split() should really only be used if you actually can have any number of components. But here it's exactly two.

i#d thus suggest to using extract_first_word() to extract the first word, and then use the rest as is.

Reply:
Switched to `extract_first_word`.